src/modm/driver/pwm/tlc594x_impl.hpp

```diff
@@ -15,40 +15,39 @@
 #endif
 
 // ----------------------------------------------------------------------------
-template<uint16_t CHANNELS, typename Spi, typename Xlat, typename Vprog, typename Xerr>
-uint8_t modm::TLC594X<CHANNELS, Spi, Xlat, Vprog, Xerr>::status[CHANNELS*3/2];
-template<uint16_t CHANNELS, typename Spi, typename Xlat, typename Vprog, typename Xerr>
-uint8_t modm::TLC594X<CHANNELS, Spi, Xlat, Vprog, Xerr>::gs[CHANNELS*3/2];
-template<uint16_t CHANNELS, typename Spi, typename Xlat, typename Vprog, typename Xerr>
-uint8_t modm::TLC594X<CHANNELS, Spi, Xlat, Vprog, Xerr>::dc[CHANNELS*3/4];
-
-// ----------------------------------------------------------------------------
-template<uint16_t CHANNELS, typename Spi, typename Xlat, typename Vprog, typename Xerr>
-void
-modm::TLC594X<CHANNELS, Spi, Xlat, Vprog, Xerr>::initialize(uint16_t channels, uint8_t dots, bool writeCH, bool writeDC)
+template<uint16_t CHANNELS, typename Spi, typename Xlat, typename Xblank, typename Vprog, typename Xerr>
+modm::TLC594X<CHANNELS, Spi, Xlat, Xblank, Vprog, Xerr>::TLC594X(uint16_t channels, uint8_t dots, bool writeCH, bool writeDC, bool enable)
 {
 	Xlat::setOutput(modm::Gpio::Low);
+	Xblank::setOutput(modm::Gpio::High);
 	Vprog::setOutput(modm::Gpio::Low);
 	Xerr::setInput();
 
-	if (dots != 0xff) setAllDotCorrection(dots, writeDC);
-	if (channels != 0xffff) setAllChannels(channels, writeCH);
+	if (channels != 0xffff) setAllChannels(channels);
+	if (dots != 0xff) setAllDotCorrection(dots);
+
+	if (writeCH) RF_CALL_BLOCKING(writeChannels());
+	if (writeDC) RF_CALL_BLOCKING(writeDotCorrection());
+
+	if (enable) this->enable();
 }
 
-template<uint16_t CHANNELS, typename Spi, typename Xlat, typename Vprog, typename Xerr>
+template<uint16_t CHANNELS, typename Spi, typename Xlat, typename Xblank, typename Vprog, typename Xerr>
 void
-modm::TLC594X<CHANNELS, Spi, Xlat, Vprog, Xerr>::latch()
+modm::TLC594X<CHANNELS, Spi, Xlat, Xblank, Vprog, Xerr>::latch()
 {
+	if (enabled) Xblank::set();
 	Xlat::set();
 	// datasheet says 20ns but that is unreliable
 	// => wait for at least 900ns
 	modm::delayMicroseconds(0.9);
```

Comment:
This function doesn't take fractionals, only integers, and this rounds down to 0. (this is my bug ;-)
Please change to
```suggestion
	modm::delayMicroseconds(1);
```

Reply:
Done.